### ISSUE TITLE
Accessibility Testing for [VAOS, VA Online Scheduling, Appointment List Redesign]

### ISSUE THREAD

**Issue by ldelacosta** [department-of-veterans-affairs/va.gov-team#53481]
### Product information

- [X] Team name, product name, and feature name have been added to the title of this issue.
- [X] Team label, product label and feature label (if applicable) have been added to this issue.

### Test notes

This test was completed on a review instance. We'll update the results when it's on staging.

Review instance: http://3bb5e6977bb1e1d270c209cda6453a37.review.vetsgov-internal/health-care/schedule-view-va-appointments/appointments/
Browser: Chrome
OS: Ventura

### Who completed the use of color and color contrast test?

@outerpress 

### Use of color and color contrast checks

- [x] All text of 20px or smaller has a 4.5:1 contrast ratio to its background (or better)
- [x] All text of 20px or larger has a 3:1 contrast ratio to its background (or better)
- [ ] Non-text elements have a 3:1  contrast ratio to their background and to neighboring elements (or better)
- [x] Color is not the only way to distinguish links from other text (eg. links are underlined)
- [x] Any charts, maps, infographics, and tables convey all information without only relying on color
- [x] Content does not refer to color, especially when providing user instructions (eg. "Click the blue button")

### How did color testing go?

✓ Text that uses #212121/rgb(33, 33, 33) **passes** at 16.1:1. This includes - 
* All pages: All headings and default text 
* Past appts: Input label and content

✓ Text that uses #004795/rgb(0, 71, 149) **passes** at 8.99:1 This includes - 
* All pages: Details and navigation links

✓ Non-text that uses #5b616b/rgb(91, 97, 107) **passes** at 6.24:1
* Past appts: Input border and icon

✓ Non-text that uses #212121/rgb(33, 33, 33) **passes** at 16.1:1
* All pages
* Modality icons

x Non-text that uses #d6d7d9/rgb(214, 215, 217) **fails** at 1.44:1
* Upcoming and past: Lighter row-divider lines

Color is not the only way to distinguish links from other text (eg. links are underlined)
✓ All pages: All links are underlined

Any charts, maps, infographics, and tables convey all information without only relying on color
✓ All pages: No charts, maps, or infographics exist.
✓ All pages: Table doesn’t rely on color to communicate information.

Content does not refer to color, especially when providing user instructions (eg. "Click the blue button")
✓ All pages: No content refers to color.

### Who completed the axe scans?

@outerpress 

### axe checks

- [x] Each page has been scanned using axe (results shared below)
- [x] axe is integrated into your end-to-end testing

### axe DevTools scan results

See below.

### How did axe testing go?

Your appointments
3 issues: id attribute value must be unique
[Upcoming axe results](https://github.com/department-of-veterans-affairs/va.gov-team/files/10711931/upcoming-appointments-3bb5e6977bb1e1d270c209cda6453a37.review.vetsgov-internal-2023-02-10.csv)

Past appointments
3 issues: id attribute value must be unique
[Past axe results](https://github.com/department-of-veterans-affairs/va.gov-team/files/10711930/past-appointments-3bb5e6977bb1e1d270c209cda6453a37.review.vetsgov-internal-2023-02-10.csv)

Pending appointments
2 issues: id attribute value must be unique
[Pending axe results](https://github.com/department-of-veterans-affairs/va.gov-team/files/10711923/pending-3bb5e6977bb1e1d270c209cda6453a37.review.vetsgov-internal-2023-02-10.csv)

### Who completed the content zoom and reflow test?

@outerpress 

### Content zoom and reflow checks

- [x] All page elements are readable and usable at 200% zoom
- [x] All page elements are readable and usable at 300% zoom
- [x] All page elements are readable and usable at 400% zoom

### How did content zoom and reflow testing go?

No issues, everything worked great.

### Who completed the keyboard navigation test?

@outerpress 

### Keyboard navigation checks

- [x] Each link, button, form input, checkbox, radio button, select menu, and custom element can receive keyboard focus
- [x] Each link, button, form input, checkbox, radio button, select menu, and custom element responds to expected keys
- [x] All elements under focus have a visible focus indicator
- [x] The order of [Tab] stops made sense and was appropriate for completing tasks

### How did keyboard testing go?

All pages were navigable according to the criteria.

However, on the upcoming appointments page the `Pending` and `Past` links are coded as buttons, which might cause confusion to screenreaders.

### Do you have any other results to share?

I ran screenreader testing using VoiceOver on MacOS Ventura. Results:

**Video 1 - upcoming appointments**

https://user-images.githubusercontent.com/2536801/218221573-56f76130-b4e6-45ee-a51f-1113bbbffaed.mov

When tabbing through using VoiceOver
* VO does not announce the time zone
* Details link doesn't announce type of appointment or modality
* VO announces 5 appointments in April when there are 8.

**Video 2 - upcoming appointments**

https://user-images.githubusercontent.com/2536801/218222144-651d0fc8-d9f4-4004-aab1-cc823d683771.mov

When navigating with ctrl+option arrows:
* Time announced as separate elements instead of the full time
* A "hidden" timezone element is announced. aria-only time zone
* Timezone announced as `MT` instead of `Mountain time`
* AM is read incorrectly. 
* Canceled appointments are not announced until the details link

**Video 3 - past appointments**

https://user-images.githubusercontent.com/2536801/218222308-38158be9-283b-4b00-9e95-5e244f07dc0a.mov

* VO doesn’t read hyphen in dropdown ranges

**Video 4 - past appointments**

https://user-images.githubusercontent.com/2536801/218222428-144bb182-8c88-472d-8934-995379313231.mov

* Details link doesn’t say it's a `past` appointment

**Video 5 - Requests**

https://user-images.githubusercontent.com/2536801/218222831-3da74af2-c3bc-4ff6-97bb-b7152d99360c.mov

* Details link should not announce a date
* Details link should say it's a `request'

**Comment by outerpress** [comment]
We also want to review the comments in the A11Y midpoint review feedback ticket: #48197 - that should help with some of the screenreader issues

**Comment by outerpress** [comment]
We have a series of tickets to handle these issues in [this Epic](https://app.zenhub.com/workspaces/vaos-team-603fdef281af6500110a1691/issues/gh/department-of-veterans-affairs/va.gov-team/38062).

**Comment by outerpress** [comment]
@ldelacosta you can close this ticket, this testing is complete.
